core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
   return(explainTuple);
 }
 
+static void appendListOfColumns(Queue* listOfColNames,ComTdb *tdb, NAString& outNAString){
+
+	if (((ComTdbHbaseAccess*)tdb)->sqHbaseTable()){// if trafodion table
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      Lng32 currPos = 0;
+	      Lng32 jj = 0;
+	      short colNameLen = *(short*)currPtr;
+	      currPos += sizeof(short);
+	      char colFam[100];
+	      while (currPtr[currPos] != ':')
+		{
+		  colFam[jj] = currPtr[currPos];
+		  currPos++;
+		  jj++;
+		}
```

Comment:
no, matter of taste. I prefer it too, but compiler will optimize the same. Some developer prefer the readability of separating, and some will even go as far as doing currPos = currPos + 1 because they dislike the ++ syntax. The snippet above is not my code, it is the code I modeled mine on, to maximize code consistency.